src/io/ipc/mod.rs

```diff
@@ -1,4 +1,80 @@
 //! APIs to read from and write to Arrow's IPC format.
+//!
+//! Inter-process communication is a method through which different processes
+//! share and pass data between them. Its use-cases include parallel
+//! processing of chunks of data across different CPU cores, transferring
+//! data between different Apache Arrow implementations in other languages and
+//! more. Under the hood Apache Arrow uses [FlatBuffers](https://google.github.io/flatbuffers/)
+//! as its binary protocol, so every Arrow-centered streaming or serialiation
+//! problem that could be solved using FlatBuffers could probably be solved
+//! using the more integrated approach that is exposed in this module.
+//!
+//! [Arrow's IPC protocol](https://arrow.apache.org/docs/format/Columnar.html#serialization-and-interprocess-communication-ipc)
+//! allows only [`RecordBatch`](crate::record_batch::RecordBatch)es or
+//! [`DictionaryBatch`](gen::Message::DictionaryBatch) to be passed
+//! around due to its reliance on a pre-defined data scheme. This limitation
+//! provides a large performance gain because serialized data will always have a
+//! known structutre, i.e. the same fields and datatypes, with the only variance
+//! being the number of rows and the actual data inside the Batch. This dramatically
+//! increases the deserialization rate, as the bytes in the file or stream are already
+//! structured "correctly".
+//!
+//! Reading and writing IPC messages is done using one of two variants - either
+//! [`FileReader`](read::FileReader) <-> [`FileWriter`](struct@write::FileWriter) or
+//! [`StreamReader`](read::StreamReader) <-> [`StreamWriter`](struct@write::StreamWriter).
+//! These two variants wrap a type `T` that implements [`Read`](std::io::Read), and in
+//! the case of the `File` variant it also implements [`Seek`](std::io::Seek). In
+//! practice it means that `File`s can be arbitrarily accessed while `Stream`s are only
+//! read in certain order - the one they were written in (first in, first out).
+//!
+//! # Examples
+//! Read and write to a file:
+//! ```
+//! use arrow2::io::ipc::{{read::{FileReader, read_file_metadata}}, {write::FileWriter}};
+//! # use std::fs::File;
+//! # use std::sync::Arc;
+//! # use arrow2::datatypes::{Field, Schema, DataType};
+//! # use arrow2::array::Int32Array;
+//! # use arrow2::record_batch::RecordBatch;
+//! // Setup the writer
+//! let path = "example.dat".to_string();
+//! let mut file = File::create(&path).unwrap();
```

Comment:
```suggestion
//! let mut file = File::create(&path)?;
```

and add an `fn main() -> Result<()>`?

imo it is important that we (Rust community) illustrate the correct use of `Result` to newcomers, since `.unwrap` is usually pretty brutal for an application difficult to catch.

Reply:
You are (of course) completely correct. I simply forgot that you can do that in doctests.